public/app/features/alerting/unified/components/rule-editor/alert-rule-form/simplifiedRouting/contactPoint/ContactPointSelector.tsx

```diff
@@ -1,90 +1,85 @@
-import { useCallback, useEffect } from 'react';
+import { QueryStatus } from '@reduxjs/toolkit/query';
+import { isEmpty } from 'lodash';
+import { useEffect } from 'react';
 import { Controller, useFormContext } from 'react-hook-form';
 
-import { SelectableValue } from '@grafana/data';
-import { ActionMeta, Field, FieldValidationMessage, Stack, TextLink } from '@grafana/ui';
+import { alertingAPI } from '@grafana/alerting/internal';
+import { ContactPointSelector as GrafanaManagedContactPointSelector } from '@grafana/alerting/unstable';
+import { config } from '@grafana/runtime';
+import { Field, FieldValidationMessage, Stack, TextLink } from '@grafana/ui';
 import { Trans, t } from 'app/core/internationalization';
-import { ContactPointSelector as ContactPointSelectorDropdown } from 'app/features/alerting/unified/components/notification-policies/ContactPointSelector';
 import { RuleFormValues } from 'app/features/alerting/unified/types/rule-form';
 import { createRelativeUrl } from 'app/features/alerting/unified/utils/url';
 
-import { ContactPointWithMetadata } from '../../../../contact-points/utils';
-
 export interface ContactPointSelectorProps {
   alertManager: string;
-  onSelectContactPoint: (contactPoint?: ContactPointWithMetadata) => void;
 }
 
-export function ContactPointSelector({ alertManager, onSelectContactPoint }: ContactPointSelectorProps) {
-  const { control, watch, trigger, setError } = useFormContext<RuleFormValues>();
+const { namespace } = config;
 
-  const contactPointInForm = watch(`contactPoints.${alertManager}.selectedContactPoint`);
+export function ContactPointSelector({ alertManager }: ContactPointSelectorProps) {
+  const { control, watch, trigger } = useFormContext<RuleFormValues>();
 
-  // Wrap in useCallback to avoid infinite render loop
-  const handleError = useCallback(
-    (err: Error) => {
-      setError(`contactPoints.${alertManager}.selectedContactPoint`, {
-        message: err.message,
-      });
-    },
-    [alertManager, setError]
-  );
+  const selectedContactPointField = `contactPoints.${alertManager}.selectedContactPoint` as const;
+  const contactPointInForm = watch(selectedContactPointField);
 
-  // if we have a contact point selected, check if it still exists in the event that someone has deleted it
-  const validateContactPoint = useCallback(() => {
-    if (contactPointInForm) {
-      trigger(`contactPoints.${alertManager}.selectedContactPoint`, { shouldFocus: true });
-    }
-  }, [alertManager, contactPointInForm, trigger]);
+  // check if the contact point still exists, we'll use listReceiver to check if the contact point exists because getReceiver doesn't work with
+  // contact point titles but with UUIDs (which is not what we store on the alert rule definition)
+  const { currentData, status } = alertingAPI.endpoints.listReceiver.useQuery({
+    fieldSelector: `spec.title=${contactPointInForm}`,
+    namespace,
+  });
```

Comment:
:+1:
Can we move this to a hook alongside the other ones in the core code? `useContactPointByTitle`?